### DISCUSSION TITLE
blobstore: allocFromBlobNodeStorages node/disk selection algorithm discussion

### DISCUSSION THREAD

**Discussion by jianwei1216** [cubefs/cubefs#3552] [Q&A] [answered]
<img width="1130" alt="图片" src="https://github.com/user-attachments/assets/7cfc1ae0-8135-4e2d-85c8-bda0327949ea">

@Cloudstriff  这个问题不大，单cluster 几百上千台机器，一次遍历很快就结束，另外这个分配过程是无锁的，效率上也没有问题

@reporter 这个选择算法中，node.FreeChunk 是不是无法作为权重了？

@Cloudstriff 可以，node的free就是权重

@reporter node.freeChunk >= randNum，才会被选中，但是 randNum 和 freeChunk 的关系似乎没有关联

**Comment by Cloudstriff** [comment, marked as answer]
这个问题不大，单cluster 几百上千台机器，一次遍历很快就结束，另外这个分配过程是无锁的，不同的分配请求间可以并发，因此效率上不会有问题。randNum是由各个Node的freeChunk累加后取随机值的，有关联

**Comment by FourSpaces** [comment]
@Cloudstriff  This algorithm performs poorly on a single node with 10 disks, often resulting in the load of some disks being 2-3 times that of other disks on the same node. It is recommended to use weighted polling when selecting disks on a single node, so that the pressure on the disks can be more balanced. It is suitable for both large and small disk scenarios.
